java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/OAuthRefreshTokenSourceAuthStrategy.java

```diff
@@ -205,9 +210,11 @@ public AccessToken refreshAccessToken() throws IOException {
             }
 
             if (shouldRefresh(this.currentToken, clock.instant())) {
+                // some jitter to minimize multiple instances refreshing at same time
+                Uninterruptibles.sleepUninterruptibly(50 + RandomUtils.nextLong(50, 200), TimeUnit.MILLISECONDS);
                 tokenResponse = exchangeRefreshTokenForAccessToken();
-                this.currentToken = asAccessToken(tokenResponse);
-                storeSharedAccessTokenIfSupported(this.currentToken);
+                AccessToken newToken = asAccessToken(tokenResponse);
+                this.currentToken = storeSharedAccessTokenIfSupported(newToken);
```

Comment:
more readable to just have `this.currentToken = newToken`, then `storedSharedAccessTokenIfSupported(newToken)` as separate line